server/src/main/java/org/opensearch/rest/action/admin/cluster/RestClusterStatsAction.java

```diff
@@ -54,7 +64,36 @@ public class RestClusterStatsAction extends BaseRestHandler {
 
     @Override
     public List<Route> routes() {
-        return unmodifiableList(asList(new Route(GET, "/_cluster/stats"), new Route(GET, "/_cluster/stats/nodes/{nodeId}")));
+        return unmodifiableList(
+            asList(
+                new Route(GET, "/_cluster/stats"),
+                new Route(GET, "/_cluster/stats/{metric}"),
+                new Route(GET, "/_cluster/stats/{metric}/{sub_metric}"),
+                new Route(GET, "/_cluster/stats/nodes/{nodeId}"),
+                new Route(GET, "/_cluster/stats/nodes/{nodeId}/{metric}"),
+                new Route(GET, "/_cluster/stats/nodes/{nodeId}/{metric}/{sub_metric}")
+            )
+        );
+    }
+
+    static final Map<String, Set<String>> METRIC_TO_SUB_METRICS_MAP;
```

Comment:
Can the metric/sub-metric map move to `ClusterStatsRequest` class ? Lets initialize all of them from a static method of Metrics class `Metrics#toSubMetricMap`

Reply:
The METRIC_TO_SUB_METRICS_MAP will not change through-out the process life, wouldn't it be better to have this initialized in the static block instead of invoking the static method.

Reply:
Yes, but can it be intiialized from a static method and the map be part of Metrics class ?